codegenerator/cli/templates/static/codegen/src/EventProcessing.res

```diff
@@ -202,12 +202,12 @@ let addEventToRawEvents = (
   let chainId = chain->ChainMap.Chain.toChainId
   let eventId = EventUtils.packEventIndex(~logIndex, ~blockNumber)
   let blockFields =
-    (block :> Types.Block.rawEventFields)->S.serializeOrRaiseWith(Types.Block.rawEventSchema)
-  let transactionFields = transaction->S.serializeOrRaiseWith(Types.Transaction.schema)
-  // Serialize to unknown, because serializing to Js.Json.t fails for Bytes Fuel type, since it has unknown schema
+    (block :> Types.Block.rawEventFields)->S.reverseConvertToJsonWith(Types.Block.rawEventSchema)
```

Comment:
What is reverseConvert?

Reply:
It's the new name for `serializeOrRaiseWith`. Which is a shorthand for `data->S.convertToJsonWith(S.reverse(schema))`

The new mental model is coming with upcoming rescript-schema V9.

Previously you could define a schema and either use it for decoding or encoding data. It's good, but not flexible enough.

The new ReScript Schema comes with the `S.reverse` function, allowing you to get the reversed version of a schema.

And then, you can apply 80 different combinations of operations to the schema.

https://x.com/dzakh_dev/status/1839406993829929031

So all schema operations are now single-directional, but you can get a reversed schema.

Reply:
Epic! I can't believe how powerful this lib is. Also I tried the ppx the other day for it and it's also so good and easy to use. I love that you can add inline attributes at a field level on records.